backend/routes/parking.py

```
@@ -0,0 +1,139 @@
+    Analyze parking sign from uploaded image
+    
+    - **file**: Image file containing parking sign
+    - **datetime_str**: Optional current datetime for context
+    
+    Returns detailed parking analysis including rules and recommendations.
+    """
+    log.info(f"Received image upload: {file.filename}, size: {file.size}")
+    
+    # Validate file
+    if not file.content_type or not file.content_type.startswith('image/'):
+        raise InvalidImageError("File must be an image")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)